cluster/examples/kubernetes/ceph/operator-with-csi.yaml

```diff
@@ -0,0 +1,574 @@
+        # current mon with a new mon (useful for compensating flapping network).
+        - name: ROOK_MON_OUT_TIMEOUT
+          value: "300s"
+        # The duration between discovering devices in the rook-discover daemonset.
+        - name: ROOK_DISCOVER_DEVICES_INTERVAL
+          value: "60m"
+        # Whether to start pods as privileged that mount a host path, which includes the Ceph mon and osd pods.
+        # This is necessary to workaround the anyuid issues when running on OpenShift.
+        # For more details see https://github.com/rook/rook/issues/1314#issuecomment-355799641
+        - name: ROOK_HOSTPATH_REQUIRES_PRIVILEGED
+          value: "false"
+        # In some situations SELinux relabelling breaks (times out) on large filesystems, and doesn't work with cephfs ReadWriteMany volumes (last relabel wins).
+        # Disable it here if you have similiar issues.
+        # For more details see https://github.com/rook/rook/issues/2417
+        - name: ROOK_ENABLE_SELINUX_RELABELING
+          value: "true"
+        # CSI enablement
+        - name: ROOK_CSI_NAMESPACE
+          value: "rook-ceph-system"        
+        - name: ROOK_CSI_ENABLE_CEPHFS
+          value: "true"
+        - name: ROOK_CSI_CEPHFS_IMAGE
+          value: "quay.io/cephcsi/cephfsplugin:v1.0.0"
+        - name: ROOK_CSI_ENABLE_RBD
+          value: "true"
+        - name: ROOK_CSI_RBD_IMAGE
+          value: "quay.io/cephcsi/rbdplugin:v1.0.0"
+        - name: ROOK_CSI_REGISTRAR_IMAGE
+          value: "quay.io/k8scsi/csi-node-driver-registrar:v1.0.2"
+          value: "quay.io/k8scsi/csi-node-driver-registrar:v1.0.2"          
+        - name: ROOK_CSI_PROVISIONER_IMAGE
+          value: "quay.io/k8scsi/csi-provisioner:canary"
+        - name: ROOK_CSI_SNAPSHOTTER_IMAGE
+          value: "quay.io/k8scsi/csi-snapshotter:canary"
+        - name: ROOK_CSI_ATTACHER_IMAGE
+          value: "quay.io/k8scsi/csi-attacher:v1.0.0"          
+        # TODO: Resizer
+
+        # Service Accounts for CephFS CSI daemons and driver
+        - name: ROOK_CSI_CEPHFS_REGISTRAR_SA
+          value: "rook-cephfs-registrar-sa"
+        - name: ROOK_CSI_CEPHFS_ATTACHER_SA
+          value: "rook-cephfs-attacher-sa"
+        - name: ROOK_CSI_CEPHFS_PROVISIONER_SA
+          value: "rook-cephfs-provisioner-sa"
+        - name: ROOK_CSI_CEPHFS_PLUGIN_SA
+          value: "rook-cephfs-plugin-sa"
+        # Service Accounts for RBD CSI daemons and driver
+        - name: ROOK_CSI_RBD_REGISTRAR_SA
+          value: "rook-cephfs-registrar-sa"
+        - name: ROOK_CSI_RBD_ATTACHER_SA
+          value: "rook-cephfs-attacher-sa"
+        - name: ROOK_CSI_RBD_PROVISIONER_SA
+          value: "rook-cephfs-provisioner-sa"
+        - name: ROOK_CSI_RBD_PLUGIN_SA
+          value: "rook-cephfs-plugin-sa"
+        # template
+        - name: ROOK_CSI_RBD_PLUGIN_TEMPLATE_PATH
+          value: "/etc/ceph-csi/rbd/csi-rbdplugin.yaml"
```

Comment:
Why do we need the templates? All other deployments/pods started by the operator are encoded with the client-go api.

Reply:
the csi driver yamls are still moving and new features are added. When things stabilize, templates will be replaced with client-go api.